src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -619,4 +619,17 @@ public function getReturnStatementsAst()
 
         return $visitor->getReturnNodes();
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
+    {
+        $objectType = (new FindTypeFromAst())->__invoke(
+            $this->reflector->getContextFactory()->createForNamespace(
```

Comment:
A `SourceLocator` should not throw any exceptions (unless something is critically broken). They must return either `Reflection*` or `null`. If it throws an exception, it stops further source location from happening. Although as a rule of thumb I throw exceptions all over the place, allowing them to be thrown breaks many combinations of chaining of aggregate/non-aggregate source locators that we have at the moment. Not sure of a nicer solution however, if you've got ideas, I'm all ears. We've already encountered all those problems in early development, hence why they return `Reflection|null` and return the `Reflection` directly; it was the cleaner solution in the end :s